core/src/main/scala/org/apache/spark/scheduler/ReplayListenerBus.scala

```diff
@@ -54,7 +55,7 @@ private[spark] class ReplayListenerBus extends SparkListenerBus with Logging {
       sourceName: String,
       maybeTruncated: Boolean = false,
       eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): Unit = {
-    val lines = Source.fromInputStream(logData).getLines()
+    val lines = Source.fromInputStream(logData, StandardCharsets.UTF_8.name()).getLines()
```

Comment:
You could avoid converting the charset to string and looking it up again by:
```scala
val lines = Source.fromInputStream(logData)(Codec.UTF8).getLines()
```

Reply:
Nice suggestion! Will address.